atomic_reactor/plugins/pre_reactor_config.py

```diff
@@ -0,0 +1,155 @@
+"""
+Copyright (c) 2017 Red Hat, Inc
+All rights reserved.
+
+This software may be modified and distributed under the terms
+of the BSD license. See the LICENSE file for details.
+"""
+from __future__ import unicode_literals
+
+from atomic_reactor.plugin import PreBuildPlugin
+
+import codecs
+import json
+import jsonschema
+import logging
+from pkg_resources import resource_stream
+import yaml
+
+
+# Key used to store the config object in the plugin workspace
+WORKSPACE_CONF_KEY = 'reactor_config'
+
+# Default configuration if none specified
+DEFAULT_CONFIG = {}
+
+
+def get_config(workflow):
+    """
+    Obtain configuration object
+    Does not fail
+
+    :return: ReactorConfig instance
+    """
+    try:
+        workspace = workflow.plugin_workspace[ReactorConfigPlugin.key]
+        return workspace[WORKSPACE_CONF_KEY]
+    except KeyError:
+        # The plugin did not run or was not successful: use defaults
+        conf = ReactorConfig(DEFAULT_CONFIG)
+        workspace = workflow.plugin_workspace.get(ReactorConfigPlugin.key, {})
+        workspace[WORKSPACE_CONF_KEY] = conf
+        workflow.plugin_workspace[ReactorConfigPlugin.key] = workspace
+        return conf
+
+
+class ClusterConfig(object):
+    """
+    Configuration relating to a particular cluster
+    """
+
+    def __init__(self, name, max_concurrent_builds=None, enabled=True):
+        self.name = str(name)
+        self.max_concurrent_builds = None
+        self.enabled = enabled
+
+        if max_concurrent_builds is not None:
+            self.max_concurrent_builds = int(max_concurrent_builds)
+
+
+class ReactorConfig(object):
+    """
+    Class to parse the atomic-reactor configuration file
+    """
+
+    CLUSTERS_KEY = 'clusters'
+
+    def __init__(self, config, log=None):
```

Comment:
I've changed that now. The DEFAULT_CONFIG is now an attribute of the ReactorConfig class.